homeassistant/components/smartthings/__init__.py

```diff
@@ -0,0 +1,201 @@
+    if broker and broker.event_handler_disconnect:
+        broker.event_handler_disconnect()
+
+    for component in SUPPORTED_PLATFORMS:
+        hass.async_create_task(hass.config_entries.async_forward_entry_unload(
+            entry, component))
+    return True
+
+
+class DeviceBroker:
+    """Manages an individual SmartThings config entry."""
+
+    def __init__(self, hass: HomeAssistantType, devices: Iterable,
+                 installed_app_id: str):
+        """Create a new instance of the DeviceBroker."""
+        self._hass = hass
+        self._installed_app_id = installed_app_id
+        self.devices = {device.device_id: device for device in devices}
+        self.event_handler_disconnect = None
+
+    async def event_handler(self, req, resp, app):
+        """Broker for incoming events."""
+        from pysmartapp.event import EVENT_TYPE_DEVICE
+
+        # Do not process events received from a different installed app
+        # under the same parent SmartApp (valid use-scenario)
+        if req.installed_app_id != self._installed_app_id:
+            return
+
+        updated_devices = set()
+        for evt in req.events:
+            if evt.event_type != EVENT_TYPE_DEVICE:
+                continue
+            device = self.devices.get(evt.device_id)
+            if not device:
+                continue
+            device.status.apply_attribute_update(
+                evt.component_id, evt.capability, evt.attribute, evt.value)
+            updated_devices.add(device.device_id)
+        _LOGGER.debug("Update received with %s events and updated %s devices",
+                      len(req.events), len(updated_devices))
+
+        async_dispatcher_send(self._hass, SIGNAL_SMARTTHINGS_UPDATE,
+                              updated_devices)
+
+
+class SmartThingsEntity(Entity):
+    """Defines a SmartThings entity."""
+
+    def __init__(self, device):
+        """Initialize the instance."""
+        self._device = device
+        self._dispatcher_remove = None
+
+    async def async_added_to_hass(self):
+        """Device added to hass."""
+        async def async_update_state(devices):
+            """Update device state."""
+            if self._device.device_id in devices:
+                await self.async_update_ha_state(True)
+
+        self._dispatcher_remove = async_dispatcher_connect(
+            self.hass, SIGNAL_SMARTTHINGS_UPDATE, async_update_state)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect the device when removed."""
+        if self._dispatcher_remove:
+            self._dispatcher_remove()
+
+    @property
+    def device_info(self):
+        """Get attributes about the device."""
+        return {
+            'identifiers': {
+                (DOMAIN, self._device.device_id)
+            },
+            'name': self._device.label,
+            'model': self._device.device_type_name,
+            'manufacturer': 'SmartThings'
```

Comment:
Is this always the case ? What is ST is linked to some Z-Wave product?

Reply:
The API doesn't expose the true manufacture -- only the device type handler (think `device driver`)--which is currently populating `model`.  I felt putting this was better than leaving it blank?

Reply:
Replaced with 'Unavailable' given the API doesn't provide this info at this time.